v2/pkg/apis/kubeflow/v2beta1/types.go

```diff
@@ -51,15 +51,6 @@ type MPIJobSpec struct {
 	// `MPIReplicaSpecs` contains maps from `MPIReplicaType` to `ReplicaSpec` that
 	// specify the MPI replicas to run.
 	MPIReplicaSpecs map[MPIReplicaType]*common.ReplicaSpec `json:"mpiReplicaSpecs"`
-
-	// MainContainer specifies name of the main container which
-	// executes the MPI code.
-	MainContainer string `json:"mainContainer,omitempty"`
-
-	// `RunPolicy` encapsulates various runtime policies of the distributed training
-	// job, for example how to clean up resources and how long the job can stay
-	// active.
-	RunPolicy *common.RunPolicy `json:"runPolicy,omitempty"`
```

Comment:
I am wondering if we should delete these fields.

Reply:
They are not used, why have them?

Reply:
Is there a design for the CRD (i.e. with explanation how these fields are meant to be used)? Do we understand that the initial reasons for adding those no longer apply?

Reply:
`MainContainer` was added to be able to start binaries using `kubectl exec`, as this requires a target container when there are more than one. This is no longer needed, as the communication happens through the SSH port.

`RunPolicy` was used in the v1alpha2 controller to configure the launcher Job retries. This is no longer used in the newer versions as we are managing plain Pods. I think we can bring it back once we move back to using Jobs. But that might take a while.

Reply:
To clarify, if we release a v1beta2 API with a field `RunPolicy` that has no effect, and then we release a new version were it suddenly has effects, that's a breaking change.
It's better to release only with the fields that have effects and add the new fields when the implementation for them is also ready.